hindsight/io_modules/derived_stream.lua

```diff
@@ -0,0 +1,377 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
General question: why `hindsight/...` for these? Some of them can be used with Heka too, right? Maybe just a `lua_modules` dir?